code-challenges.test.js

```diff
@@ -10,33 +10,124 @@
+
+    22 | // Create a describe statement 
+//     23 | describe ("messageCoder", () => {
+//   > 24 |     it("takes in a function and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.",  () =>{
+//        |                                                                                                    ^
+//     25 |         const secrectCodeWord1 = "Lackadaisical"
+//     26 |         const secretCodeWord2 = "Gobbledygook"
+//     27 |         const secretCodeWord3 = "Eccentric"
+
+//     Test Suites: 1 failed, 1 total
+//     Tests:       0 total
+//     Snapshots:   0 total
+//     Time:        0.875 s
+//     Ran all test suites matching /code-challenges.test.js/i.
+//     error Command failed with exit code 1.
+
 
 // b) Create the function that makes the test pass.
 
-// --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
-// a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
 
-const letterA = "a"
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+// 2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
+
+
+// const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+
+// const letterA = "a"
 // Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
+// const letterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
+// a) Create a test with expects statement using the variable provided.
+describe("vowelFavorerer", ()=> {
+    it("Takes in an array of words and a single letter and returns an array of all the words containing that particular letter",()=>{
+        
+        const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+        const letterA = "a"
+        const letterE = "e"
+
+        expect(vowelFavorerer(letterA)).toEqual("Mango", "Apricot", "Peach")
+        expect(vowelFavorerer(letterE)).toEqual("Cherry", "Blueberry", "Peach")
+    
+    })
+ })
 // b) Create the function that makes the test pass.
 
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
+//Psuedo code 
+//For this code I can use a similar method to how I did ruby challenge 1. I'll be using sort & include so I can filter out the words based on what I want it to include.
+ const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+  function vowelFavorerer(fruitArray){
+    const vowelFavorerer = fruitArray.filter(fruitArray => fruitArray.includes('e')).sort();
+    return vowelFavorerer
+  }
```

Comment:
No need to use `.sort()`, as that will actually throw off your output from what you want. Again, as I stated in the last comment, this function should be taking 2 arguments, so there should be 2 parameters made when you are declaring your function. Also, this is currently only checking the `fruitArray` for whether each word includes specifically the letter 'e', so you will never be able to run this function to check if a word includes the letter 'a'. Also, make sure your variable names are different from your function name to avoid confusion, and it is also best practice to use arrow syntax for functions like we teach in class. When you find code online, you can convert it to arrow syntax.